blaze/expr/collections.py

```diff
@@ -235,6 +250,47 @@ def tail(child, n=10):
     return Tail(child, n)
 
 
+class Sample(Expr):
+    """Random row-wise sample.  Can specify `n` or `frac` for an absolute or
+    fractional number of rows, respectively.
+
+    Examples
+    --------
+    >>> from blaze import symbol
+    >>> accounts = symbol('accounts', 'var * {name: string, amount: int}')
+    >>> accounts.sample(n=2).dshape
+    dshape("var * {name: string, amount: int32}")
+    >>> accounts.sample(frac=0.1).dshape
+    dshape("var * {name: string, amount: int32}")
+    """
+    __slots__ = '_hash', '_child', 'n', 'frac'
+
+    def _dshape(self):
+        return self._child.dshape
+
+    def __str__(self):
+        arg = 'n={}'.format(self.n) if self.n is not None else 'frac={}'.format(self.frac)
+        return '%s.sample(%s)' % (self._child, arg)
+
+
+@copydoc(Sample)
+def sample(child, n=None, frac=None):
+    if n is frac is None:
+        raise TypeError("sample() missing 1 required argument, 'n' or 'frac'.")
+    if n is not None and frac is not None:
+        raise ValueError("n ({}) and frac ({}) cannot both be specified.".format(n, frac))
+    if n is not None:
+        n = int(n)
```

Comment:
Should we maybe use `operator.index` to reject things like `float` and `Decimal`?